src/plugins/discover/public/application/components/discover_grid/discover_value_with_filter.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { ReactNode, useState } from 'react';
+import { EuiButtonIcon, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { DocViewFilterFn, ElasticSearchHit } from '../../doc_views/doc_views_types';
+import { IndexPattern } from '../../../../../data/common/index_patterns/index_patterns';
+
+/**
+ * Draft component displaying filter lens icons on hover with the possibility to add a filter
+ */
+export const DiscoverGridValueWithFilter = ({
+  value,
```

Comment:
Draft component to provide mouse hovered lens icons for adding filters like in the legacy table

![image](https://user-images.githubusercontent.com/463851/85863928-1f033b80-b7c4-11ea-8aeb-1100898755c6.gif)

works but has some problems when text is truncated and the expand icon is displayed e.g. in Firefox

![image](https://user-images.githubusercontent.com/463851/85864554-f760a300-b7c4-11ea-8d23-52701381529e.gif)



The question here is if we could/should add this to the grids expand icon (making it exandable), and I think there should be an option to display this icon(s) on mouse hover, what do you think dear @chandlerprall

Reply:
Just a note that we should be using `plusInCircle` and `minusInCircle` icons here to match what we have in the sidebar and in Lens.

Also, wondering if we really need to keep the filter in/out buttons in the cell? Seems like on a mockup shared last [year](https://github.com/elastic/kibana/issues/38646#issuecomment-543209904) we didn't have it. See: 

![67019899-5bc25300-f0cb-11e9-8a5b-6dee3b23a8f8](https://user-images.githubusercontent.com/4016496/86055370-81e91280-ba10-11ea-918d-d4f840a800eb.gif)

Reply:
The reason for this is, that adding a filter in the new DataGrid needed 3 click, first selecting the cell, clicking on expand, then adding the filter. In the legacy grid it was just a single click

![image](https://user-images.githubusercontent.com/463851/86105400-93d5be80-babf-11ea-943b-65f3f7ef2673.gif)

I've suggested a PR to allow the expanding of a cell with a single click btw: https://github.com/elastic/eui/pull/3668

Reply:
I'm excited to use the same way in the Data table visualization: https://github.com/elastic/kibana/issues/64444

Reply:
@sulemanof great, so we should align this, will push some changes here this week, do you already have hands on it?